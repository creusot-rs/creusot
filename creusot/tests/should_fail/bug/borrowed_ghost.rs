extern crate creusot_contracts;
use creusot_contracts::*;

pub fn use_borrowed() {
    let mut x = gh! { true };
    let r = &mut x; // x = ?, r = (gh true, x)
    *r = gh! { !x.inner() }; // r = (gh (not (inner x)), x)
                             // resolve r: x = gh (not (inner x))
    proof_assert! { x.inner() == !x.inner() } // UNSOUND!
}
