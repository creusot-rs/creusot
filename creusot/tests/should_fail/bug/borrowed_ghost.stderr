error[creusot]: Use of borrowed variable x
 --> borrowed_ghost.rs:7:10
  |
7 |     *r = gh! { !x.inner() }; // r = (gh (not (inner x)), x)
  |          ^^^^^^^^^^^^^^^^^^
  |
  = note: this error originates in the macro `gh` (in Nightly builds, run with -Z macro-backtrace for more info)

error: aborting due to previous error

